src/grading-settings/assignment-section/index.jsx

```diff
@@ -0,0 +1,223 @@
+              />
+              <AssignmentItem
+                className="course-grading-assignment-number-droppable"
+                title={intl.formatMessage(messages.numberOfDroppableTitle)}
+                descriptions={intl.formatMessage(messages.numberOfDroppableDescription)}
+                type="number"
+                min={MIN_NUMBER_VALUE}
+                errorMsg={intl.formatMessage(messages.numberOfDroppableErrorMessage)}
+                name={dropCount}
+                gradeField={gradeField}
+                value={gradeField.dropCount}
+                onChange={(e) => handleAssignmentChange(e, idx)}
+                secondErrorMsg={(
+                  <FormattedMessage
+                    id="course-authoring.grading-settings.assignment.number-of-droppable.second.error.message"
+                    defaultMessage="Cannot drop more {type} assignments than are assigned."
+                    values={{ type: gradeField.type }}
+                  />
+                )}
+                errorEffort={errorList[`${dropCount}-${idx}`]}
+              />
+            </ol>
+            {gradeField.minCount !== courseAssignmentUsage?.length && Boolean(courseAssignmentUsage?.length) && (
+              <AlertMessage
+                className="course-grading-assignment-item-alert-warning"
+                variant="warning"
+                icon={Warning}
+                title={(
+                  <FormattedMessage
+                    id="course-authoring.grading-settings.assignment.alert.warning.usage.title"
+                    defaultMessage="Warning: The number of {type} assignments defined here does not match the current number of {type} assignments in the course:"
+                    values={{ type: gradeField.type }}
+                  />
+                )}
+                description={(
+                  <>
+                    <span className="course-grading-assignment-item-alert-warning-list-label">
+                      {courseAssignmentUsage.length} Final assignment(s) found:
+                    </span>
+                    <ol className="course-grading-assignment-item-alert-warning-list">
+                      {courseAssignmentUsage.map(assignmentItem => (
+                        <li key={assignmentItem}>{assignmentItem}</li>
+                      ))}
+                    </ol>
+                  </>
+                )}
+                aria-hidden="true"
+              />
+            )}
+            {!courseAssignmentUsage?.length && Boolean(gradeField.type) && (
+              <AlertMessage
+                className="course-grading-assignment-item-alert-warning"
+                variant="warning"
+                icon={Warning}
+                title={(
+                  <FormattedMessage
+                    id="course-authoring.grading-settings.assignment.alert.warning.title"
+                    defaultMessage="Warning: The number of {type} assignments defined here does not match the current number
```

Comment:
the same as above

Reply:
<img width="1225" alt="image" src="https://github.com/raccoongang/frontend-app-course-authoring/assets/93188219/dd97a87c-f241-4e6e-b717-e4adcc760b85">

Reply:
Added